src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnosticsForSpan.cs

```diff
@@ -376,7 +376,11 @@ public async Task<bool> TryGetAsync(ArrayBuilder<DiagnosticData> list, Cancellat
 
                 var analyzers = stateSets.SelectAsArray(stateSet => stateSet.Analyzer);
                 var analysisScope = new DocumentAnalysisScope(_document, span, analyzers, kind);
-                var executor = new DocumentAnalysisExecutor(analysisScope, _compilationWithAnalyzers, _owner._diagnosticAnalyzerRunner, _logPerformanceInfo);
+
+                // We log performance info and mark the request as high priority when we have a blocking request for a sub-span
+                // in document, i.e. for lightbulb code path for Ctr + Dot invocation.
+                var executor = new DocumentAnalysisExecutor(analysisScope, _compilationWithAnalyzers, _owner._diagnosticAnalyzerRunner,
+                    highPriority: _isBlockingRequestForSubSpanInDocument, logPerformanceInfo: _isBlockingRequestForSubSpanInDocument);
```

Comment:
i have mixed feelings.  instead of trying to infer this, it feels nicer to just thread this value all the way up teh caller stack and have the feature just say what it is.  IMO at least...

Reply:
Sounds good, I personally felt that threading it to the callers is better. I'll make this change.